core/src/test/scala/unit/kafka/metrics/MetricsTest.scala

```diff
@@ -75,6 +75,20 @@ class MetricsTest extends KafkaServerTestHarness with Logging {
     assertEquals(metrics.keySet.asScala.count(_.getMBeanName == "kafka.server:type=KafkaServer,name=ClusterId"), 1)
   }
 
+  @Test
+  def testGeneralBrokerTopicMetricsAreGreedilyRegistered(): Unit = {
+    val topic = "test-broker-topic-metric"
+    createTopic(topic, 2, 1)
+
+    // The broker metrics for all topics should be greedily registered
+    assertTrue("General topic metrics don't exist", topicMetrics(None).nonEmpty)
```

Comment:
I'd replace this assert with one that checks one of the map entries.

Reply:
What is the reasoning?

Reply:
An assert that fails saying that some metrics should be there but aren't is less useful than an assert that tells you that a particular metric doesn't exist. Generally, strive to avoid `assertTrue` if there is an alternative since tests that fail in this way tend to be the most annoying. The first thing you have to do is change the assert to get more information.

Reply:
Similar for the `size` thing, if you know what the elements should be, it's better to assert that the collections are the same rather than just checking the `size`. These are guidelines and don't always apply, but something to keep in mind.

Reply:
Thanks for the suggestion. I agree and I'll keep these guidelines in mind next time. In this particular instance, it seems hard to do the proper assertion as:
1) MeterWrapper encapsulates the yammer `Meter` object, so unless we make the lazyMeter variable public, we can't really tell whether it's initialized or not (calling meter() would initialize it)
2) We can only access the `MetricName` instance from the default registry (topicMetrics() helper) and our internal map hosts the yammer `Meter`, which doesn't expose names. It's hard to reconcile both

Reply:
Makes sense, thanks for looking into it.